src/main/java/io/kestra/plugin/notion/AbstractNotionTask.java

```diff
@@ -0,0 +1,180 @@
+        )
+        private URI uri;
+
+        @Schema(
+            title = "Success",
+            description = "Whether the operation was successful"
+        )
+        private Boolean success;
+
+        @Schema(
+            title = "Message",
+            description = "Operation result message"
+        )
+        private String message;
+    }
+
+    /**
+     * Helper method to extract page title from Notion page properties
+     */
+    protected String extractPageTitle(Map<String, Object> properties) {
+        if (properties == null) {
+            return null;
+        }
+
+        // Notion stores title in different formats, try common patterns
+        Map<String, Object> titleProp = (Map<String, Object>) properties.get("title");
+        if (titleProp == null) {
+            titleProp = (Map<String, Object>) properties.get("Name");
+        }
+        
+        if (titleProp != null && titleProp.get("title") instanceof java.util.List) {
+            java.util.List<?> titleArray = (java.util.List<?>) titleProp.get("title");
+            if (!titleArray.isEmpty() && titleArray.get(0) instanceof Map) {
+                Map<String, Object> firstTitle = (Map<String, Object>) titleArray.get(0);
+                Object plainText = firstTitle.get("plain_text");
+                if (plainText != null) {
+                    return plainText.toString();
+                }
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Helper method to build common output from NotionResponse.
+     * Returns a builder that can be further customized before calling build().
+     * 
+     * @param response the NotionResponse containing page data
+     * @return Output builder with common fields populated
+     */
+    protected Object buildCommonOutput(NotionResponse response) {
+        return Output.builder()
+            .pageId(response.getId())
+            .url(response.getUrl())
+            .title(extractPageTitle(response.getProperties()))
+            .createdTime(response.getCreatedTime())
+            .lastEditedTime(response.getLastEditedTime())
+            .archived(response.getArchived())
+            .properties(response.getProperties())
+            .success(true);
+    }
+
+    /**
+     * Helper method to build error output
+     */
+    protected Output buildErrorOutput(String message) {
+        return Output.builder()
+            .success(false)
+            .message(message)
+            .build();
+    }
+
+    /**
+     * Validates that the pageId property is provided and not empty
+     */
+    protected String validateAndRenderPageId(RunContext runContext) throws Exception {
+        String renderedPageId = runContext.render(this.pageId).as(String.class).orElse("");
+        
+        if (renderedPageId.isEmpty()) {
+            throw new IllegalArgumentException("pageId is required and cannot be empty");
+        }
```

Comment:
```suggestion
        String renderedPageId = runContext.render(this.pageId).as(String.class).orElseThrow();
```

This will do the same.
Moreover as there is a NotNull annotation this would never happen when used in a flow.